src/main/java/Duke.java

```diff
@@ -1,10 +1,96 @@
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constuctor for Duke
+     */
+    public Duke() {
+        this.storage = new Storage();
+        this.ui = new Ui();
+        this.storage = new Storage();
+        ui.preload();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            if (e.getMessage().equals("LoadFileError")) {
+                ui.loadFileError();
+            }
+        }
+    }
+
+    /**
+     * Method to get response to an input for GUI Duke Program.
+     *
+     * @param input input command.
+     * @return Response String output.
+     * @throws DukeException Different kinds of Duke Exception errors.
+     *
+     */
+    public String getResponse(String input) throws DukeException {
+        try {
+            Command c = Parser.parse(input);
+            return c.execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            if (e.getMessage().equals("IndexOutOfBound")) {
+                return ui.arrayIndexOutOfBoundsError();
+            }
+            if (e.getMessage().equals("EmptyIndex")) {
+                return ui.emptyIndexError();
+            }
+            if (e.getMessage().equals("EmptyDescription")) {
+                return ui.emptyDescriptionError(e.getInfo());
+            }
+            if (e.getMessage().equals("EmptyDetails")) {
+                return ui.emptyDetailsError(e.getInfo());
+            }
+            if (e.getMessage().equals("UnknownCommand")) {
+                return ui.unknownCommandError();
+            }
+            if (e.getMessage().equals("TypeNotFound")) {
+                return ui.typeNotFound();
+            }
+        }
+        return "Error! Please Try Another Command!";
+    }
+
+    /**
+     * Runs the program for the Text-Based Duke Program.
+     */
+    public void run() {
+        ui.introduction();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                if (e.getMessage().equals("IndexOutOfBound")) {
+                    ui.arrayIndexOutOfBoundsError();
+                }
+                if (e.getMessage().equals("EmptyIndex")) {
+                    ui.emptyIndexError();
+                }
+                if (e.getMessage().equals("EmptyDescription")) {
+                    ui.emptyDescriptionError(e.getInfo());
+                }
+                if (e.getMessage().equals("EmptyDetails")) {
+                    ui.emptyDetailsError(e.getInfo());
+                }
+                if (e.getMessage().equals("UnknownCommand")) {
+                    ui.unknownCommandError();
+                }
+            }
```

Comment:
I like how easy it is to read your code! It is also good that you avoided deep nesting and having an arrow head structure. You did this practice earlier as well so good job!